src/ORM/DataObject.php

```diff
@@ -3242,11 +3240,11 @@ public function debug(): string
      *   - it still returns an object even when the field has no value.
      *   - it only matches fields and not methods
      *   - it matches foreign keys generated by has_one relationships, eg, "ParentID"
+     *   - if the field exists, the return value is ALWAYS a DBField instance
      *
-     * @param string $fieldName Name of the field
-     * @return DBField The field as a DBField object
+     * Returns null if the field doesn't exist
      */
-    public function dbObject($fieldName)
+    public function dbObject(string $fieldName): ?DBField
```

Comment:
It always did return null in some situations (see the unchanged implementation below). Now we're just being explicit  about it.